bin/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+#---------------------------------------------------------------------------#
+# Copyright (c) 2024 Alexey Kokoshnikov <alexeikokoshnikov@nil.foundation>
+#
+# Distributed under the Boost Software License, Version 1.0
+# See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt
+#---------------------------------------------------------------------------#
+
+cmake_minimum_required(VERSION 3.5)
+
+cm_find_package(CM)
+include(CMDeploy)
+
+if(NOT CMAKE_CROSSCOMPILING)
+    find_package(Boost COMPONENTS REQUIRED filesystem log log_setup program_options thread system)
+elseif(CMAKE_CROSSCOMPILING)
+    if(NOT TARGET boost)
+        include(ExternalProject)
+        set(Boost_LIBRARIES boost_random)
+        externalproject_add(boost
+                            PREFIX ${CMAKE_BINARY_DIR}/libs/boost
+                            GIT_REPOSITORY git@github.com:boostorg/boost.git
+                            GIT_TAG boost-1.77.0
+                            BUILD_IN_SOURCE TRUE
+                            CMAKE_ARGS -DCMAKE_CROSSCOMPILING_EMULATOR=${CMAKE_CROSSCOMPILING_EMULATOR} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                            BUILD_COMMAND cmake --build . --target ${Boost_LIBRARIES}
+                            INSTALL_COMMAND "")
+    else()
+        set(Boost_LIBRARIES boost_random)
+    endif()
+endif()
+
+if (NOT WITNESS_COLUMNS)
+    set(WITNESS_COLUMNS 15 CACHE STRING "Number of witness columns")
+endif()
+if (NOT PUBLIC_INPUT_COLUMNS)
+    set(PUBLIC_INPUT_COLUMNS 1 CACHE STRING "Number of public input columns")
+endif()
+if (NOT COMPONENT_CONSTANT_COLUMNS)
+    set(COMPONENT_CONSTANT_COLUMNS 5 CACHE STRING "Number of component constant columns")
+endif()
+if (NOT LOOKUP_CONSTANT_COLUMNS)
+    set(LOOKUP_CONSTANT_COLUMNS 30 CACHE STRING "Number of lookup constant columns")
+endif()
+if (NOT COMPONENT_SELECTOR_COLUMNS)
+    set(COMPONENT_SELECTOR_COLUMNS 50 CACHE STRING "Number of lookup selector columns")
+endif()
+if (NOT LOOKUP_SELECTOR_COLUMNS)
+    set(LOOKUP_SELECTOR_COLUMNS 6 CACHE STRING "Number of lookup selector columns")
+endif()
+if (NOT SECURITY_PARAMETER_LAMBDA)
+    set(SECURITY_PARAMETER_LAMBDA 9 CACHE STRING "Number of FRI queries")
+endif()
+if (NOT SECURITY_PARAMETER_GRINDING_BITS)
+    set(SECURITY_PARAMETER_GRINDING_BITS 0 CACHE STRING "Number of FRI grinding bits")
+endif()
+
+
+add_definitions(-DWITNESS_COLUMNS=${WITNESS_COLUMNS})
```

Comment:
I would rather recommend creating a header by `configure_file` command and then include it into the dependent files. As I understand, the header could already contain parameters in `struct BaseParametersPolicy {constexpr Key = Value; ... }` format, so there will be no need to list them again in the tools.

Reply:
Not sure about header: 
1. now setting values in `CMakeLists` - common way for all repos (should be enabled soon in `proof_producer`), so it easy to be alligned.
2. Automatically generated headers - looks complex.
In long term perspective we want to do these parameters dynamic, so it temporary solution which will be removed.

Reply:
OK, approving as a temporary solution. But I still don't think that passing 5+ parameters as definitions directly to compiler is a good idea. That's what config files are for.